quic/s2n-quic-events/events/common.rs

```diff
@@ -734,6 +734,32 @@ enum DatagramDropReason {
     PathLimitExceeded,
     /// The peer initiated a connection migration without supplying enough connection IDs to use.
     InsufficientConnectionIds,
+    /// Application provided invalid MTU configuration.
+    MtuValidation {
+        /// MTU configuration for the endpoint
+        endpoint_mtu_config: MtuConfig,
+        // TODO expose connection level mtu config.
+        // error from mtu::Config
+        // TODO expose the remote address
+        // remote_addr: SocketAddress<'a>,
```

Comment:
I found these values non-trivial and didnt want to delay this PR more. I can add issues if we decide we need these values. The Config might not be trivial to get since we dont return it on failed validation.

Reply:
https://github.com/aws/s2n-quic/issues/2254